### PR TITLE
add osx support

### PR DESCRIPTION
**Pull request by tduehr**

### CODE REVIEW

File: lib/rake_compiler_dock.rb [hunk 1]
```diff
@@ -58,11 +58,16 @@ def exec(*args)
...
+    case RUBY_PLATFORM
+    when /mingw|mswin/
       # Change Path from "C:\Path" to "/c/Path" as used by boot2docker
       pwd = Dir.pwd.gsub(/^([a-z]):/i){ "/#{$1.downcase}" }
       uid = 1000
       gid = 1000
+    when /darwin/
+      pwd = Dir.pwd
+      uid = 1000
+      gid = Process.gid
```
Comment (larskanis) [inline lib/rake_compiler_dock.rb]: Thank you for working on this! I would like to understand the change. Why is the uid set to a fixed value but the git of the host system?

Comment (tduehr) [inline lib/rake_compiler_dock.rb]: I didn't need to change it to get it to work. However, it might be better correct to just change it.

**Comment by larskanis** [comment]
Does this solve all issues you had with building ffi on rake-compiler-dock on OSX? I mean: is is equally usable as on Linux?

**Comment by tduehr** [comment]
Yes and I assume so. Most of my problems were because I hadn't used docker before. There's still some tweaks that could be done but this works. I didn't try it in linux but it doesn't change anything on linux either.

**Pull request merged**